src/main/java/de/zalando/aruha/nakadi/config/NakadiConfig.java

```diff
@@ -86,18 +79,36 @@ public ZooKeeperHolder zooKeeperHolder() {
     }
 
     @Bean
+    @Profile("kafka")
     public KafkaLocationManager getKafkaLocationManager() {
         return new KafkaLocationManager();
     }
 
     @Bean
+    @Profile("kafka")
     public KafkaFactory kafkaFactory() {
         return new KafkaFactory(getKafkaLocationManager());
     }
 
     @Bean
-    public KafkaRepositorySettings topicRepositorySettings() {
+    @Profile("kafka")
+    public KafkaRepositorySettings kafkaRepositorySettings() {
         return new KafkaRepositorySettings();
     }
 
+    @Bean
+    public TopicRepository topicRepository() {
+        if (environment.acceptsProfiles("kafka")) {
```

Comment:
Ok, if you all think it's not yet time for that - fine for me.
I removed "kafka" profile now